assets/js/callback.js

```diff
@@ -1 +1,37 @@
-alert("authentication callback complete.")
+// alert("authentication callback complete.")
+
+function redirectUser() {
+
+  let redirectUrl
+
+  const params = new URLSearchParams(window.location.search)
+  const referringUrl = decodeURIComponent(params.get("ref"))
+  // const hasProject = decodeURIComponent(params.get("projects"))
+  const userToken = decodeURIComponent(params.get("access-token"))
+
+  if (referringUrl) {
+    redirectUrl = referringUrl
+  } else if ( userHasProject(userToken)) {
+    redirectUrl = "xyz/dashboard/projects"
+  } else {
+    redirectUrl = "/"
```

Comment:
This and the last one might be connected. Perhaps the index page checks for projects. Also this wouldn't be "/" because you would need to be in an authenticated area. I think it will be in TPEN-interfaces, but we can look into that later.

Reply:
it makes more sense to do the check on the index page since we want to spend the least time possible on the callback page. once we get this working, I can easily move the code TPEN_interfaces if we decide its best there. I have refactored it to work from anywhere though